Resolve #818 -- AddProjectile function expanded

lzardy opened the pull request:
Fixes #818. Has x and y as spawn parameters for cases where projectile is spawned by client-side spell or when projectile should spawn at casted position rather than owner position.

lzardy committed:
AddProjectileCoords function

Fixes #818. Has x and y as spawn parameters for cases where projectile is spawned by client-side spell or when projectile should spawn at casted position rather than owner position.

lzardy committed:
Fix Defaults pointer

woops

danil179 commented on GameServerCore/Domain/ISpell.cs:
```diff
@@ -31,6 +31,7 @@ public interface ISpell: IUpdate
         void ApplyEffects(IAttackableUnit u, IProjectile p);
         void LevelUp();
         void AddProjectile(string nameMissile, float toX, float toY, bool isServerOnly = false);
```
I suggest you to change the implementation one of the function to use the other one `AddProjectile` will use your new `AddProjectileCoords`.

danil179 reviewed (commented):
Reviewed.

lzardy committed:
AddProjectileCoords->AddProjectile

No use in having AddProjectile if AddProjectileTarget accomplishes the same thing; they both are hard coded to spawn at owner.X, owner.Y, so rename AddProjectileCoords to AddProjectile.

lzardy committed:
Update Submodule Pointer
